src/rgw/rgw_rest_s3.cc

```diff
@@ -6134,11 +6134,21 @@ const char* RGWSelectObj_ObjStore_S3::header_value_str[3] = {"Records", "applica
 RGWSelectObj_ObjStore_S3::RGWSelectObj_ObjStore_S3():
   s3select_syntax(std::make_unique<s3selectEngine::s3select>()),
   m_s3_csv_object(std::unique_ptr<s3selectEngine::csv_object>()),
+  m_s3_parquet_object(std::unique_ptr<s3selectEngine::parquet_object>()),
   m_buff_header(std::make_unique<char[]>(1000)),
   chunk_number(0),
+  m_s3select_query(""),
+  m_parquet_type(false),
   crc32(std::unique_ptr<boost::crc_32_type>())
 {
   set_get_data(true);
+  m_rgw_api = new s3selectEngine::rgw_s3select_api;
```

Comment:
I feel it again here that we (or the s3electEngine) could have already allocated whatever the engine is handing us here.

Reply:
no need to allocate each request.
(but since it incomplete type)
 
since we wish to avoid including s3select-header into the RGW header, need to add a dedicated header for rgw-s3select.

Reply:
it is allocated on the stack.